darwin/datatypes.py

```diff
@@ -565,9 +560,18 @@ def make_polygon(
     Annotation
         A polygon ``Annotation``.
     """
+
+    if darwin_v1:
+        polygon_data = {"path": point_path}
+    else:
+        # Lets handle darwin V2 datasets
+        if not isinstance(point_path[0], list):
+            point_path = [point_path]
+        polygon_data = {"paths": point_path}
+
     return Annotation(
         AnnotationClass(class_name, "polygon"),
-        _maybe_add_bounding_box_data({"path": point_path}, bounding_box),
+        _maybe_add_bounding_box_data(polygon_data, bounding_box),
```

Comment:
This doesn't look correct to me. You seem to be putting v1 vs v2 differences into abstract model that doesn't care about versions. 
Can we use the `datatypes.make_polygon` or `datatypes.make_complex_polygon` instead? I'm 99% sure that putting `paths` (plural) on a class of type `polygon` is invalid in our codebase. 

For reference you can take a look at the logic that reads JSON files directly here: https://github.com/v7labs/darwin-py/blob/79fd606063dff08c1e22d08c452b75f663cc25a2/darwin/utils/utils.py#L647
As you can see this code uses both `polygon` and `complex_polygon` classes for internal structures and this is how it should stay.

Reply:
Thank you @rslota , it's I that misunderstood the  internal darwin format and assume that it should adhere to the darwin-v1 and v2 formats. Will make changes to the PR that keeps the internal format intact, thanks!